src/views/Submit.vue

```diff
@@ -262,11 +289,32 @@ export default {
 			this.$refs.submitButton.click()
 		},
 
+		/*
+		 * Methods for catching unwanted unload events
+		 */
+		beforeWindowUnload(e) {
+			if (this.confirmStayInEditedForm()) {
+				// Cancel the window unload event
+				e.preventDefault()
+				e.returnValue = ''
+			}
+		},
+		confirmLeavingPage() {
+			return confirm(t('forms', 'You have unsaved changes! Do you still want to leave?'))
+		},
+		confirmStayInEditedForm() {
+			return (
+				!this.submitForm
+                && !this.confirmLeavingPage()
```

Comment:
This will even show the dialog if there are *no* unsaved changes, even if you simply in the submit view of your own form for previewing and want to switch back to edit view.
I would suggest to check if there are any answers already like this

```suggestion
				!this.submitForm
				&& Object.keys(this.answers).length > 0
				&& !this.confirmLeavingPage()
```